### PR TITLE
Add remainder cleanup for BitArray equality comparison

### LINKED ISSUE #39328: Contents of BitArray padding shouldn't influence ==(A, B)
Because `==(A::BitArray, B::BitArray)` [compares](https://github.com/JuliaLang/julia/blob/68133407404f3e1221385d495b47eb8585b258af/base/bitarray.jl#L1156-L1159) the UInt64 packed chunks comprising the argument arrays, modifying the contents of the padding in the last chunks causes `==` to return `false`, even though the arrays are element-wise equal:
```
julia> v1 = BitVector([true, false])
2-element BitVector:
 1
 0

julia> v2 = copy(v1);

julia> v1 == v2
true

julia> v2.chunks[1] += 16
0x0000000000000011

julia> v1 == v2
false

julia> all(v1 .== v2)
true
```

### PR DESCRIPTION
**Pull request by stillyslalom**
Closes #39328

### CODE REVIEW

**Comment by chriselrod** [comment]
```julia
julia> using Random

julia> a = bitrand(2000);

julia> b = bitrand(2000);

julia> function eq(A::BitArray, B::BitArray)
           size(A) != size(B) && return false
               A.chunks == B.chunks && return true

           # clean up remainder
           n_rem = length(A) - 64*(length(A.chunks) - 1)
           a, b = A.chunks[end], B.chunks[end]
           c = (UInt64(1) << n_rem) - 1
           (a & c) == (b & c)
       end
eq (generic function with 1 method)

julia> eq(a,b)
false

julia> b.chunks[end] = last(a.chunks);

julia> eq(a,b)
true
```

**Comment by chriselrod** [comment]
But then it won't fix https://github.com/JuliaLang/julia/issues/39328

**Comment by chriselrod** [comment]
Perhaps:
```julia
@view(A.chunks[begin:end-1]) == @view(B.chunks[begin:end-1]) || return false
```

**Comment by stillyslalom** [comment]
I was trying to avoid `view`s after being scarred in earlier releases. Thanks for the reminder that they're fast now!

File: base/bitarray.jl [hunk 1]
```diff
@@ -1155,9 +1155,14 @@ end
...
     size(A) != size(B) && return false
-    return A.chunks == B.chunks
-end
+    A.chunks == B.chunks || return false
 
+    # clean up remainder
+    n_rem = length(A) - 64*(length(A.chunks) - 1)
+    a, b = A.chunks[end], B.chunks[end]
+    c = (UInt64(1) << n_rem) - 1
+    (a & c) == (b & c)
```
Comment (mbauman) [inline base/bitarray.jl:1164]:
```suggestion
    A.chunks[end] & _msk_end(A) == B.chunks[end] & _msk_end(B)
```

In addition, of course, to Chris' review.

**Comment by chriselrod** [comment]
I do think from the discussion in the issue that we should probably require that the trailing digits be `0` (and say that packages shouldn't be modifying them).
Between `==`, reductions, and `find` methods, most use cases are currently taking advantage of the `0` digits, and I think if we are going to change it, it should probably be all at once. And only done after actually establishing that that is actually worth it (and I suspect it isn't).

**Comment by vtjnash** [comment]
The issue this was attempting to fix was deemed to be invalid

**Pull request closed without merging**